src/main/java/seedu/address/logic/parser/wallet/WalletDeleteCommandParser.java

```diff
@@ -0,0 +1,32 @@
+package seedu.address.logic.parser.wallet;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.wallet.WalletDeleteCommand;
+import seedu.address.logic.parser.Parser;
+import seedu.address.logic.parser.ParserUtil;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new DeleteCommand object
+ */
+public class WalletDeleteCommandParser implements Parser<WalletDeleteCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the DeleteCommand
+     * and returns a DeleteCommand object for execution.
+     *
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public WalletDeleteCommand parse(String args) throws ParseException {
+        try {
+            Index index = ParserUtil.parseIndex(args);
+            return new WalletDeleteCommand(index);
```

Comment:
I think return should be outside try-catch block

Reply:
I followed the implementation for PeopleDeleteCommandParser though. They have the return inside try block. Should I follow them or move it out?

Reply:
I think either way is fine